src/components/Registration/Manual/SearchAttendee.vue

```diff
@@ -1,148 +1,269 @@
+onMounted(() => {
+  refs.value = Array(showInfo.length)
+    .fill(null)
+    .map(() => ref(null))
+})
+function setRef(index) {
+  return (el) => {
+    refs.value[index] = el
+  }
 }
 
+const menuOpen = ref(false)
+
+const toggleMenu = () => (menuOpen.value = !menuOpen.value)
+
+const query = ref('')
+const typing = ref(false)
+
 watch(query, () => {
   if (query.value !== '') typing.value = true
   else typing.value = false
 })
 
-const checkIn = (item) => {
-  item.checkedIn.value = true
-}
-
 const items = [
   {
     id: 1,
     name: 'Wei Tat Chung',
-    position: 'Chairperson',
-    role: 'Organiser',
     email: 'wtc@email.com',
-    organisation: 'SUSS AI-IG',
-    checkedIn: ref(false)
+    checkedIn: ref(false),
+    info: {
+      role: 'Chairperson',
+      memberType: 'Organiser',
+      organisation: 'SUSS AI-IG'
+    }
   },
   {
     id: 2,
-    name: 'Shaun Laclemence',
-    position: 'Chairperson',
-    role: 'Organiser',
+    name: 'Don Chia',
     email: 'wtc@email.com',
-    organisation: 'SUSS AI-IG',
-    checkedIn: ref(false)
+    checkedIn: ref(false),
+    info: {
+      role: 'Chairperson',
+      memberType: 'Organiser',
+      organisation: 'SUSS AI-IG'
+    }
   },
   {
     id: 3,
-    name: 'Don Chia',
-    position: 'Chairperson',
-    role: 'Organiser',
+    name: 'Shaun Ming Laclemence',
     email: 'wtc@email.com',
-    organisation: 'SUSS AI-IG',
-    checkedIn: ref(false)
+    checkedIn: ref(false),
+    info: {
+      role: 'Chairperson',
+      memberType: 'Organiser',
+      organisation: 'SUSS AI-IG'
+    }
   },
   {
-    id: 3,
-    name: 'Don Chia',
-    position: 'Chairperson',
-    role: 'Organiser',
+    id: 4,
+    name: 'Very very very very very very very very very very long name',
     email: 'wtc@email.com',
-    organisation: 'SUSS AI-IG',
-    checkedIn: ref(false)
+    checkedIn: ref(false),
+    info: {
+      role: 'Chairperson',
+      memberType: 'Organiser',
+      organisation: 'SUSS AI-IG'
+    }
   },
   {
-    id: 3,
-    name: 'Don Chia',
-    position: 'Chairperson',
-    role: 'Organiser',
+    id: 5,
+    name: 'Wei Tat Chung',
     email: 'wtc@email.com',
-    organisation: 'SUSS AI-IG',
-    checkedIn: ref(false)
+    checkedIn: ref(false),
+    info: {
+      role: 'Chairperson',
+      memberType: 'Organiser',
+      organisation: 'SUSS AI-IG'
+    }
   }
 ]
+
+const showInfo = [
+  {
+    id: 'info1',
+    name: 'Role',
+    show: ref(true),
+    action: () => (showInfo[0].show.value = !showInfo[0].show.value)
+  },
+  {
+    id: 'info2',
+    name: 'Member Type',
+    show: ref(true),
+    action: () => (showInfo[1].show.value = !showInfo[1].show.value)
+  },
+  {
+    id: 'info3',
+    name: 'Organisation',
+    show: ref(true),
+    action: () => (showInfo[2].show.value = !showInfo[2].show.value)
+  }
+]
+
+const filtered = ref(false)
+
+const updateMenu = () => {
+  if (menuOpen.value == true) {
+    refs.value[0].checked = showInfo[0].show.value
```

Comment:
can i know what is this for? it probably could have been a computed property or something?

Reply:
This updates the checked states of each option whenever the menu is opened

Reply:
and whether or not 1 or more options is unchecked (menu is filtered) which determines whether or not the filled funnel icon is used or outline funnel